bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolEmuEMThingHandler.java

```diff
@@ -0,0 +1,299 @@
+        } else {
+            logger.debug("No available bridge handler found yet. Bridge: {} .", bridge.getUID());
+        }
+        return bridgeHandler;
+    }
+
+    private void registerResolThingListener(@Nullable ResolBridgeHandler bridgeHandler) {
+        if (bridgeHandler != null) {
+            bridgeHandler.registerResolThingListener(this);
+        } else {
+            logger.debug("Can't register {} at bridge as bridgeHandler is null.", this.getThing().getUID());
+        }
+    }
+
+    public int getVbusAddress() {
+        return vbusAddress;
+    }
+
+    public void useConnection(Connection connection) {
+        EmDeviceEmulator device = this.device;
+        if (device != null) {
+            device.stop();
+            device.removePropertyChangeListener(this);
+        }
+        device = new EmDeviceEmulator(connection, deviceId);
+        this.device = device;
+        device.addPropertyChangeListener(this);
+        device.start();
+        for (int i = 1; i <= 5; i++) {
+            setRelayChannelValue(i, device.getRelayValueByNr(i));
+        }
+        startAutomaticUpdate();
+    }
+
+    public void stop() {
+        EmDeviceEmulator device = this.device;
+        if (device != null) {
+            device.stop();
+        }
+        ScheduledFuture<?> updateJob = this.updateJob;
+        if (updateJob != null) {
+            updateJob.cancel(false);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String chID = channelUID.getId();
+        boolean update = false;
+        int channel = chID.charAt(chID.length() - 1) - '0';
+        float value = 0;
+        int intValue = 0;
+
+        if (command instanceof QuantityType<?>) {
+            value = Objects.requireNonNullElse(((QuantityType<?>) command).toUnit(SIUnits.CELSIUS),
+                    new QuantityType<>(888.8, SIUnits.CELSIUS)).floatValue();
+            update = true;
+        } else if (command instanceof OnOffType) {
+            intValue = ((OnOffType) command).equals(OnOffType.ON) ? 1 : 0;
+            update = true;
+        } else if (command instanceof DecimalType) {
+            intValue = ((DecimalType) command).intValue();
+            value = intValue;
+            update = true;
+        } else {
+            update = false;
+        }
+
+        if (update) {
+            EmDeviceEmulator dev = device;
+            if (dev != null) {
+                if (chID.startsWith(CHANNEL_TEMP)) {
+                    dev.setResistorValueByNrAndPt1000Temperatur(channel, value);
+                    updateState(channelUID, new DecimalType(value));
+                } else if (chID.startsWith(CHANNEL_SWITCH)) {
+                    if (intValue == 0) {
+                        /* switch is open => 1 megaohm */
+                        dev.setResistorValueByNr(channel, 1000000000);
+                        updateState(channelUID, OnOffType.OFF);
```

Comment:
Why update the channels here? at this point it gets a command because the value of the channel changed. So this update changes the value to the value the channel already contains? So it seems redundant to do this.

Reply:
I added these updates, because otherwise the MainUI did not correctly update the view (of the widget I tried with). Maybe I just did something wrong as I am not yet used to MainUI or whatever. Let me know if I should remove it.

Reply:
@Hilbrand any remark on that? - I have added the update here to have the MainUI showing the update. Should I remove it?